File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/OpenFeedingCloseCommand.cs
using SIE.Domain;
using SIE.MES.TaskManagement.Dispatchs;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.TaskManagement.Dispatchs.Commands
{
    /// <summary>
    /// 打开上料命令
    /// </summary>
    public class OpenFeedingCloseCommand : ViewCommand<List<double>>
    {
        protected override object Excute(List<double> args, string scope)
        {
            RT.Service.Resolve<DispatchController>().OpenFeedingClose(args);
            return true;
        }
    }
}

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/OpenFeedingCloseCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Dispatchs.Commands.OpenFeedingCloseCommand', {
    meta: { text: "打开上料", group: "edit", iconCls: "icon-ModifyPassword icon-blue" },
    canExecute: function (view) {
        var selecteditems = view.getSelection();
        if (selecteditems != null && selecteditems.length > 0 && selecteditems.all(p => p.getIsFeedingClose() == true)) {
            return true;
            //for (var i = 0; i < selecteditems.length; i++) {
            //    if (selecteditems[i].data.TaskStatus === 40 || selecteditems[i].data.TaskStatus === 50 || selecteditems[i].data.TaskStatus === 60) {
            //        return false;
            //    }
            //}
            //return true;
        }
        return false;
    },
    execute: function (view, source) {
        view.execute({
            data: view.getSelectionIds(),
            success: function (res) { //回调
                view.reloadData();
            }
        });
    },
});